<!-- -->
<launch>

<!-- carla parameter -->
    <arg name='host' default='localhost'/> <!-- host the carla server is running on -->
    <arg name='port' default='2000'/> <!-- port of the carla server -->
    <arg name='timeout' default='50'/> <!-- the carla timeout for server communication -->
    <arg name="role_name" default="hero"/> <!-- role name of the ego vehicle (used for identification in many nodes) -->
    <arg name="goal_pose" default=" "/>

        <!-- Provide a route and access to the CARLA waypoint API (required by carla_ad_agent) -->
    <node pkg="carla_waypoint_publisher" type="carla_waypoint_publisher_mapp.py" name="carla_waypoint_publisher_mapp_$(arg role_name)" output="screen">
        <param name="role_name" value="$(arg role_name)" />
        <param name="host" value="$(arg host)" />
        <param name="port" value="$(arg port)" />
        <param name="timeout" value="$(arg timeout)" />
        <param name="goal_pose" value="$(arg goal_pose)" />
    </node>

    <node pkg="mapp" type="kill.py" name="kill_$(arg role_name)" output="screen" required="true">
        <param name="role_name" value="$(arg role_name)"/>
    </node>

</launch>